pkg/admission/customresourcevalidation/image/validate_image.go

```diff
@@ -0,0 +1,92 @@
+package image
+
+import (
+	"fmt"
+	"io"
+
+	configv1 "github.com/openshift/api/config/v1"
+
+	"github.com/openshift/origin/pkg/admission/customresourcevalidation"
+	"k8s.io/apimachinery/pkg/api/validation"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"k8s.io/apiserver/pkg/admission"
+)
+
+// Register registers a plugin
+func Register(plugins *admission.Plugins) {
+	plugins.Register("config.openshift.io/ValidateImage", func(config io.Reader) (admission.Interface, error) {
+		return customresourcevalidation.NewValidator(configv1.Resource("images"), validateObjCreateFn, validateObjUpdateFn, validateObjStatusUpdateFn)
```

Comment:
Would be better as an interface rather than passing raw functions around, unless you're going to expect people to define these inline.

Reply:
> Would be better as an interface rather than passing raw functions around, unless you're going to expect people to define these inline.

Yeah, and people probably want/need a slice of them to handle versions.